source/agora/utils/TxBuilder.d

```diff
@@ -253,6 +254,16 @@ public struct TxBuilder
         return this;
     }
 
+    /// Sign with a given key and append ubytes if given
+    public static Unlock signWithSpecificKey (KeyPair key, ubyte[] append = null) (in Transaction tx, in OutputRef out_ref)
+        @safe nothrow
+    {
+        auto pair = SigPair(key.sign(tx.getChallenge()), SigHash.All);
+        if (append)
+            return Unlock(toPushOpcode(pair[]) ~ append);
+        return Unlock(toPushOpcode(pair[]));
+    }
```

Comment:
@omerfirmak so this is ok?

Reply:
It should work as long as `append` is a valid script. If it is raw data if should still be passed to `toPushOpcode()`. So the caller should be aware of what it is passing as `append`